ass/__init__.py

```diff
@@ -56,3 +64,520 @@
+           format.
+
+        :param data: string to parse
+        """
+        ass_process_codec_private(self.track, encode_str(data), len(data))
+
+    def ass_process_chunk(self, data, timecode, duration):
+        """Parse a chunk of subtitle stream data.
+
+        :param data: string to parse
+        :param timecode: starting time of the event (milliseconds)
+        :param duration: duration of the event (milliseconds)
+        """
+        ass_process_chunk(
+            self.track,
+            encode_str(data),
+            len(data),
+            int(timecode),
+            int(duration),
+        )
+
+    def ass_set_check_readorder(self, check_readorder):
+        """Set whether the ReadOrder field when processing a packet with
+           ass_process_chunk() should be used for eliminating duplicates.
+
+        :param check_readorder: 0 means do not try to eliminate duplicates; 1 means
+         use the ReadOrder field embedded in the packet as unique identifier, and
+         discard the packet if there was already a packet with the same ReadOrder.
+         Other values are undefined.
+        """
+        ass_set_check_readorder(self.track, int(check_readorder))
+
+    def ass_flush_events(self):
+        """Flush buffered events.
+        """
+        ass_flush_events(self.track)
+
+    def ass_read_styles(self, fname, codepage):
+        """Read styles from file into already initialized track.
+
+        :param fname: file name
+        :param codepage: encoding (iconv format)
+        """
+        if ass_read_styles(
+            self.track, get_encoded_path(fname), encode_str(codepage)
+        ):
+            raise LibassError("Can't read styles from {}".format(fname))
+
+    def ass_step_sub(self, now, movement):
+        """Calculates timeshift from now to the start of some other subtitle
+           event, depending on movement parameter.
+
+        :param now: current time in milliseconds
+        :param movement: how many events to skip from the one currently displayed
+        """
+        return ass_step_sub(self.track, int(now), int(movement))
+
+    def __getattr__(self, name):
+        track = self.track[0]
+
+        class EmptyDict:
+            pass
+
+        return_value = {
+            "name": cast(track.name, c_char_p).value,
+            "styles": track.styles[0],
+            "events": track.events[0],
+            "style_format": cast(track.style_format, c_char_p).value,
+            "event_format": cast(track.event_format, c_char_p).value,
+            "Language": cast(track.Language, c_char_p).value,
+        }.get(name, EmptyDict())
+
+        if isinstance(return_value, EmptyDict):
+            return getattr(track, name)
+
+        return return_value
+
+    def __del__(self):
+        """Deallocate track and all its child objects (styles and events).
+        """
+        if self.track:
+            ass_free_track(self.track)
```

Comment:
Fat `__init__.py` is an antipattern, let's move this to a separate file and optionally import these classes here
I haven't read these carefully but these look like bubblesub shenanigans. Do we need them here in this package?

Reply:
This package wasn't created thinking at `bubblesub`. I followed `pyffms2` implementation. it's all inside `__init__`. How do you propose to split the class? Each class is a new file?

Reply:
I mean the whole renderer class.
>Each class is a new file
Pretty much, yes, unless there are a ton of small classes then no. Basically common sense.
ffms2 has only garbage in `__init__.py` because we didn't bother to clean it up.

Reply:
I agree

Reply:
Unfortunately I wasn't able to remove the `from .enums import *` because we need that for the `__all__` list. Do you know how to remove that?